ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt

```diff
@@ -0,0 +1,43 @@
+package kotlinx.coroutines.javafx
+
+import javafx.beans.value.ChangeListener
+import javafx.beans.value.ObservableValue
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.conflate
+
+/**
+ * Creates an instance of a cold [Flow] that subscribes to the given [ObservableValue] and produces
+ * its values as they change.
+ *
+ * The resulting flow is conflated, meaning that if several values arrive in a quick succession, only
+ * the last one will be produced.
+ *
+ * It produces at least one value.
+ *
+ * Since this implementation uses [ObservableValue.addListener], even if this [ObservableValue]
+ * supports lazy evaluation, eager computation will be enforced while the flow is being collected.
+ */
+@ExperimentalCoroutinesApi
+fun <T: Any> ObservableValue<T>.asFlow(): Flow<T> = callbackFlow<T> {
+    val listener = ChangeListener<T> { observable, oldValue, newValue ->
+        try {
+            offer(newValue)
+        } catch (e: CancellationException) {
```

Comment:
What is going to happen if  `offer` throws a non-cancellation exception, e.g. because `collect {}` block has thrown it? Is it behaviour consistent and non-racy?

Reply:
As usual for this PR, I think that this is fine because everything here (this offer here, `addListener`, `removeListener` and so on) happens in the same JavaFX thread. So it's probably fully consistent and not racy.

I found a problem though: if someone runs `collect` in the JavaFX main thread, everything hangs. Should we consider this case? And can we?

Reply:
Removing `runBlocking`, as recommended above, did solve the hanging problem.